Sources/NIO/BSDSocketAPI.swift

```diff
@@ -0,0 +1,228 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if os(Windows)
+import let WinSDK.IPPROTO_IP
+import let WinSDK.IPPROTO_IPV6
+import let WinSDK.IPPROTO_TCP
+import let WinSDK.SOL_SOCKET
+
+import let WinSDK.IP_ADD_MEMBERSHIP
+import let WinSDK.IP_DROP_MEMBERSHIP
+import let WinSDK.IP_MULTICAST_IF
+import let WinSDK.IP_MULTICAST_LOOP
+import let WinSDK.IP_MULTICAST_TTL
+
+import let WinSDK.IPV6_JOIN_GROUP
+import let WinSDK.IPV6_LEAVE_GROUP
+import let WinSDK.IPV6_MULTICAST_HOPS
+import let WinSDK.IPV6_MULTICAST_IF
+import let WinSDK.IPV6_MULTICAST_LOOP
+import let WinSDK.IPV6_V6ONLY
+
+import let WinSDK.TCP_NODELAY
+
+import let WinSDK.SO_ERROR
+import let WinSDK.SO_KEEPALIVE
+import let WinSDK.SO_LINGER
+import let WinSDK.SO_RCVBUF
+import let WinSDK.SO_RCVTIMEO
+import let WinSDK.SO_REUSEADDR
+import let WinSDK.SO_REUSE_UNICASTPORT
+#elseif os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
+import Darwin
+#else
+import Glibc
+#endif
+
+public enum BSDSocket {
+}
+
+public extension BSDSocket {
```

Comment:
We are trying to avoid using access modifiers on extensions, and prefer to put them on the individual declarations (it makes it easier to evaluate what the actual access modifier is for a given declaration when you don't have to pop the stack, especially for large extensions).

This applies to all extensions added here.